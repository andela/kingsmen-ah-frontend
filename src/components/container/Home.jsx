import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Header from '@components/commons/Header';

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    const { user, profile, isAuthenticated } = this.props;

    return (
      <div className='bg-gray-100 font-sans w-full min-h-screen m-0'>
        <Header
          user={{ user: { ...user, isAuthenticated } }}
          profile={profile}
        />
      </div>
    );
  }
}

Home.propTypes = {
  user: PropTypes.shape({}).isRequired,
  profile: PropTypes.shape({}).isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  errors: PropTypes.shape({}).isRequired
};

const mapStateToProps = state => ({
  user: state.auth.user,
  profile: state.auth.profile,
  isAuthenticated: state.auth.isAuthenticated,
  errors: state.auth.errors
});

export default connect(
  mapStateToProps,
  {}
)(Home);
