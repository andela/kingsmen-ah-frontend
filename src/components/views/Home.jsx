import React from 'react';
import Header from '@components/commons/Header';

const user = {
  isAuthenticated: true,
  username: 'Macco'
};

const profile = {
  firstname: 'Emmanuel',
  lastname: 'Okwara',
  avatar: './public/img/logo.png'
};

export default function Home() {
  return (
    <div className='bg-gray-100 font-sans w-full min-h-screen m-0'>
      <Header user={user} profile={profile} />
    </div>
  );
}
