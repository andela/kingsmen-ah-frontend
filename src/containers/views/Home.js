import { connect } from 'react-redux';
import Home from '@components/views/Home';
import { getAllArticles } from '@actions/articles';

const mapStateToProps = state => ({
  isLoading: state.article.loading,
  articles: state.article.articles,
  errors: state.article.errors,
});

export default connect(mapStateToProps, { getAllArticles })(Home);