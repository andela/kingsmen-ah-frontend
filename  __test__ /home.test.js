import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Home from '@components/views/Home';

configure({ adapter: new Adapter() });

const setup = () => {
  const wrapper = shallow(<Home />);
  return wrapper;
};

describe('<Home />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should display This is the Home page', () => {
    const text = wrapper.find('h1').text();
    expect(text).toEqual('This is the Home page');
  });

  it('should render the Home component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
