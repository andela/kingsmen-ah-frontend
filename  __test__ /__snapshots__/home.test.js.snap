// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render the Home component correctly 1`] = `
<div>
  <h1>
    This is the Home page
  </h1>
</div>
`;
